doc/threadsafe_function.md

```diff
@@ -0,0 +1,219 @@
+```
+
+- `[in] value`: The `napi_threadsafe_function` which is a handle for an existing napi threadsafe function.
+
+Returns a non-empty `Napi::ThreadSafeFunction` instance.
+
+### New
+
+Creates a new instance of the `Napi::ThreadSafeFunction` object. The `New` function has several overloads for the various optional parameters: skip the optional parameter for that specific overload.
+
+```cpp
+New(napi_env env,
+    const Function& callback,
+    const Object& resource,
+    ResourceString resourceName,
+    size_t maxQueueSize,
+    size_t initialThreadCount,
+    DataType* data,
+    Finalizer finalizeCallback,
+    Context* context);
+```
+
+- `env`: The `napi_env` environment in which to construct the `Napi::ThreadSafeFunction` object.
+- `callback`: The `Function` to call from another thread.
+- `[optional] resource`: An object associated with the async work that will be passed to possible async_hooks init hooks.
+- `resourceName`: A JavaScript string to provide an identifier for the kind of resource that is being provided for diagnostic information exposed by the async_hooks API.
+- `maxQueueSize`: Maximum size of the queue. `0` for no limit.
+- `initialThreadCount`: The initial number of threads, including the main thread, which will be making use of this function.
+- `[optional] data`: Data to be passed to `finalizeCallback`.
+- `[optional] finalizeCallback`: Function to call when the `ThreadSafeFunction` is being destroyed.  This callback will be invoked on the main thread when the thread-safe function is about to be destroyed. It receives the context and the finalize data given during construction (if given), and provides an opportunity for cleaning up after the threads e.g. by calling `uv_thread_join()`. It is important that, aside from the main loop thread, there be no threads left using the thread-safe function after the finalize callback completes. Must implement `void operator()(Env env, DataType* data, Context* hint)`, skipping `data` or `hint` if they are not provided.
+- `[optional] context`: Data to attach to the resulting `ThreadSafeFunction`. Can be retreived via `GetContext()`.
+
+Returns a non-empty `Napi::ThreadSafeFunction` instance.
+
+### Acquire
+
+Add a thread to this thread-safe function object, indicating that a new thread will start making use of the thread-safe function. 
+
+```cpp
+bool Napi::ThreadSafeFunction::Acquire()
+```
+
+Returns `true` if the thread can successfully be added to the thread count.
+
+### Release
+
+Indicate that an existing thread will stop making use of the thread-safe function. A thread should call this API when it stops making use of this thread-safe function. Using any thread-safe APIs after having called this API has undefined results in the current thread, as it may have been destroyed.
+
+```cpp
+bool Napi::ThreadSafeFunction::Release()
+```
+
+Returns `true` if the thread-safe function has successfully released.
```

Comment:
"... has been successfully released.",  maybe?

Reply:
Addressed; likewise with the signature change mentioned above.